### PR TITLE
validation script for docker SE added

### PR DESCRIPTION
**Pull request by sergeikiktev**
@guillaume-brossard-adsk 
@david-richer-adsk 
Please check(CentOS 7 only).
Some check can be joined in loop but good as first version.

### CODE REVIEW

File: setup/setup_validation_docker.sh [hunk 1]
```diff
@@ -0,0 +1,231 @@
+#!/bin/bash
+
+#######################################################################################
+#  This scripts is doing basic docket on Centos 7 only  #
```
Comment (david-richer-adsk) [inline setup/setup_validation_docker.sh]: docker

File: setup/setup_validation_docker.sh [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        export OVER="OK"
+    fi
+    printf "RAM                    : %-10s %-10sKB\n" $OVER $MEM
+
+    # SELinux check
+    export OVER="OK"
+    if [ ! \( `getenforce` == "Permissive" -o `getenforce` == "Disabled" \) ]; then
+      export OVER="BAD"
+    fi
+    printf "SE Linux               : %-10s %-10s\n" $OVER `getenforce`
```
Comment (david-richer-adsk) [inline setup/setup_validation_docker.sh]: Do we really care about SE Linux with Docker SE?

**Comment by david-richer-adsk** [comment]
Look great!

**Comment by david-richer-adsk** [comment]
May be it should validate by default if we run without options?

File: setup/setup_validation_docker.sh [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    validate
+    shift
+    ;;
+--install)
+    install_all
+    shift
+    ;;
+*)
+    ;;
+esac
```
Comment (guillaume-brossard) [inline setup/setup_validation_docker.sh]: No option should default to displaying help.

File: setup/setup_validation_docker.sh [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+    if (( $(echo "$CCV >= $DRV" |bc -l) )); then
+	export OVER="OK"
+    else
+	export OVER="BAD"
+    fi
+    printf "Kernel version         : %-10s %-10s\n" $OVER $CCV
+
+    # CPU model
+    CPU=`cat /proc/cpuinfo |grep "model name" |awk -F: '{print $2}'| uniq`
```
Comment (guillaume-brossard) [inline setup/setup_validation_docker.sh:63]: Indentation is wrong for CPU Model and Hostname.
```
Kernel version         : OK         3.10
CPU model              : Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
```

I expected:
```
Kernel version         : OK         3.10
CPU model              :                Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
```

File: setup/setup_validation_docker.sh [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    done
+    printf "S3 available           : %-10s\n" $OVER
+
+
+    # DATABASE INFO
+    echo -e "\nDATABASE INFO"
+    echo "==============="
+    export OVER="NO"
+    if (rpm -qa|grep postgres); then
+    STATUS=`systemctl status postgresql.service|grep active|awk '{print $2}'`
```
Comment (guillaume-brossard) [inline setup/setup_validation_docker.sh]: In my case, postgresql is running and your script says it is not. Here is how I installed it:
https://docs.google.com/document/d/1OI8s9sFDLoqhc0J_E05-Gu7wEWW3n9Q6bhutZmCaXF4/edit#

File: setup/setup_validation_docker.sh [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+	fi
+    else
+	echo "Postgres is not installed."
+    fi
+    printf "PostgreSQL-running     : %-10s %-10s\n" $OVER $STATUS
+
+    # Postgres parameters
+    if sudo test -f $PG_FILE; then 
+	printf "\nPostgres parameters:\n"
+	sudo cat /var/lib/pgsql/9.*/data/postgresql.conf|grep -E 'shared_buffers|work_mem|maintenance_work_mem|vacuum_cost_delay|effective_cache_size|max_connections|statement_timeout'|uniq
```
Comment (guillaume-brossard) [inline setup/setup_validation_docker.sh]: Why not $PG_FILE?

Comment (guillaume-brossard) [inline setup/setup_validation_docker.sh]: The `*` doesn't work out for me.

**Review by guillaume-brossard — CHANGES_REQUESTED** [review]
Very nice! It will be very useful.

File: setup/setup_validation_docker.sh (hunk 2 above)
Comment (sergeikiktev) [reply setup/setup_validation_docker.sh]: Usually docker comes with additional SELinux policy package and will not work correct if SELinux enabled and package missing. Should we make a note in documentation then?
e.g.:
[shotgun@localhost ~]$ rpm -qa|grep docker
docker-ce-selinux-17.03.1.ce-1.el7.centos.noarch
docker-ce-17.03.1.ce-1.el7.centos.x86_64

File: setup/setup_validation_docker.sh (hunk 5 above)
Comment (sergeikiktev) [reply setup/setup_validation_docker.sh]: I Could not get installation script from documentation.
Also CentOS 7 is using "systemctl status" command by default.
I also found that postgres is installing to "/var/lib/pgsql/data" by default.

Works on my side.

screen shot 2017-06-19 at 15 39 29

File: setup/setup_validation_docker.sh (hunk 6 above)
Comment (sergeikiktev) [reply setup/setup_validation_docker.sh]: Fixed...

Comment (sergeikiktev) [reply setup/setup_validation_docker.sh]: fixed.

**Comment by sergeikiktev** [comment]
Fixed. Please review

File: setup/setup_validation_docker.sh (hunk 2 above)
Comment (guillaume-brossard) [reply setup/setup_validation_docker.sh]: @david-richer-adsk ?

File: setup/setup_validation_docker.sh [hunk 7]
```diff
@@ -0,0 +1,239 @@
+#!/bin/bash
+
+###########################################################################
+#  This scripts is doing basic server environment check on Centos 7 only  #
+###########################################################################
+
+set +e
+
+#Postgresql configuration file
+PG_FILE=/var/lib/pgsql/data/postgresql.conf
```
Comment (guillaume-brossard) [inline setup/setup_validation_docker.sh]: Default is `/var/lib/pgsql/data/<version>/postgresql.conf` isn't?

File: setup/setup_validation_docker.sh [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+    printf "S3 available           : %-10s\n" $OVER
+
+
+    # DATABASE INFO
+    echo -e "\nDATABASE INFO"
+    echo "==============="
+    export OVER="NO"
+    PGCHECK=`rpm -qa|grep postgres`
+    if [ -n "$PGCHECK" ];then
+    STATUS=`systemctl status postgresql.service|grep active|awk '{print $2}'`
```
Comment (guillaume-brossard) [inline setup/setup_validation_docker.sh]: Hey Sergei, like I said, this is how we suggest to client to install PostgreSQL. This is the instruction I followed, on a VM, to test your script. So either we need to change the installation script, or make sure the script can detect PostgreSQL running when installed as our script is suggesting.

https://github.com/shotgunsoftware/docker-se/tree/master/postgresql

Comment (guillaume-brossard) [reply setup/setup_validation_docker.sh]: In my case, the following is working:
`sudo systemctl status postgresql-9.3.service`

**Review by guillaume-brossard — CHANGES_REQUESTED** [review]

File: setup/setup_validation_docker.sh (hunk 2 above)
Comment (david-richer-adsk) [reply setup/setup_validation_docker.sh]: I just don't know if Docker and our containers will work or not.  Let's play safe and require SELinux to be disabled on host like before

File: setup/setup_validation_docker.sh (hunk 8 above)
Comment (sergeikiktev) [reply setup/setup_validation_docker.sh]: Hm... Weird, default postgres comes with centos repo does not contain version number in /var/lib/pgsql/ and "sudo systemctl status postgresql-9.3.service".

Ok, all good then.

**Comment by david-richer-adsk** [comment]
Fixed postgresql detection tested with postgres 9.3 and 9.6

**Pull request merged**